src/initiate.js

```diff
@@ -82,7 +105,7 @@ export const initiateRendering = async (
   if (localMbtilesMatches && !sourceDir) {
     const msg =
       "Stylesheet has local mbtiles file sources, but no sourceDir is set";
-    throw new Error(msg);
+    return handleError(new Error(msg), "checking for local mbtiles sources");
```

Comment:
This is an example of a bad request, right?  The caller has to change something before this will ever succeed.

I might define (in utils.js) two separate error handlers, one for bad client request, one for unexpected server error.  The two would set different errorCode (as you have it now) or different status (per an idea I share in src/utils.js).

A minor variation of this is to define two different Error subclasses and you only `throw BadRequestError(msg)` here, leaving the entry point (clip.js or azure_queue_service.js) to catch that error if it wants.  Example in Python:
- error definitions: https://github.com/ConservationMetrics/cmi-jobs/blob/1391e0751719b24d1f6a88f518dc653ae1dde309/azure-functions/http_utils.py#L13-L26
- core logic throws exception, not handles it: https://github.com/ConservationMetrics/cmi-jobs/blob/1391e0751719b24d1f6a88f518dc653ae1dde309/azure-functions/launch_tasks/launch_tasks_azfn.py#L120-L123